### PR TITLE
Bugfix/increase tokens

### PR DESCRIPTION
**Pull request by jlopezlira**
This PR can help prevent the 400 error when exceeding tokens by giving the option to increase it by the user from his configuration file.

Related Bug: https://github.com/di-sukharev/opencommit/issues/120

### CODE REVIEW

**Comment by codeagencybe** [comment]
@di-sukharev 
any reason this PR not available yet? Seems like this might solve the problem wiht the 400 error code many people have?

**Comment by jlopezlira** [comment]
Hi! @di-sukharev @codeagencybe 

My solution is based on allowing the user to increase the token size, here is a good article about token size https://medium.com/@russkohn/mastering-ai-token-limits-and-memory-ce920630349a


Another solution will be updating our current prompt, but is currently working good!

Thanks,

JJ

**Comment by codeagencybe** [comment]
> Hi! @di-sukharev @codeagencybe
> 
> My solution is based on allowing the user to increase the token size, here is a good article about token size https://medium.com/@russkohn/mastering-ai-token-limits-and-memory-ce920630349a
> 
> Another solution will be updating our current prompt, but is currently working good!
> 
> Thanks,
> 
> JJ

ok so that means we have to wait until @di-sukharev has merged your PR?

**Comment by jlopezlira** [comment]
That's correct 👍

File: .github/workflows/codeql.yml
Comment (di-sukharev) [inline .github/workflows/codeql.yml:1]: why is this file highlighted, its already in dev, could you pls manage

File: .github/workflows/dependency-review.yml
Comment (di-sukharev) [inline .github/workflows/dependency-review.yml:1]: also this file

File: README.md
Comment (di-sukharev) [inline README.md:160]: this is also in dev, why is that highlighted?

**Review by di-sukharev — CHANGES_REQUESTED** [review]
fix the `.github/workflows/*` files pls, they should not be here

**Comment by di-sukharev** [comment]
conflict

File: .github/workflows/codeql.yml
Comment (jlopezlira) [reply .github/workflows/codeql.yml:1]: I was not updating my branch, thanks! Great catch.

File: .github/workflows/dependency-review.yml
Comment (jlopezlira) [reply .github/workflows/dependency-review.yml:1]: Same as 🔝

File: README.md
Comment (jlopezlira) [reply README.md]: Same as 🔝

**Comment by jlopezlira** [comment]
Hey there! Any chance to merge this PR?

**Review by di-sukharev — APPROVED** [review]

**Comment by di-sukharev** [comment]
thanks everybody

**Comment by codeagencybe** [comment]
@di-sukharev is there a list of all the official feature flags? 
When I run oc config --help it only tells me the structure, but not which options I can set. 
If I visit the GH page here, there is no official list either. 

I see that the last merge now has new options available, but after updating to latest version 2.0.10 (it does not show 2.0.11 weirdly), my ~/.opencommit file does not show the new options either. 

From the commit, I can see there are 2 new config options added? 
```
let basePath = config?.OPENAI_BASE_PATH;
let maxTokens = config?.OPENAI_MAX_TOKENS;

```

Does that mean my config like this:

```
OPENAI_API_KEY=sk-xxxxxx
emoji=true
description=true
```

I now need to manually change to below? 

```
OPENAI_API_KEY=sk-xxxxxx
emoji=true
description=true
OPENAI_BASE_PATH=??
OPENAI_MAX_TOKENS=??
```

Or did the upgrade go wrong and should it automatically add those new base config fields? 
If you could please elaborate or even better, update the project readme to clarify this better. 

Thanks

**Comment by di-sukharev** [comment]
done @codeagencybe

**Comment by codeagencybe** [comment]
@di-sukharev 
Thanks! I see it only reflected on the npm page itself, not here in Github ;) 
But anyway, I got it now and updated. 

I did notice a new bug that got introduced now. 

I ran the command to set max tokens. After that I ran the command again to set a higher value. 
Then you git this error:

image

To fix this, you have to edit the ~/.opencommit and remove the line, save. 
Then run the oc set config command again to set the higher or lower value you want.

**Comment by Jossec101** [comment]
I'm having issues whatever token I put on macOS, 

`Unsupported config key OPENAI_MAX_TOKENS: Must be a number``

My config:

```
OPENAI_API_KEY=****
OPENAI_MAX_TOKENS=1000
```

**Pull request merged**
